Explorer/Assets/DCL/Character/CharacterMotion/Animation/AvatarAnimationEventsHandler.cs

```diff
@@ -46,22 +46,50 @@ public class AvatarAnimationEventsHandler : MonoBehaviour
         [PublicAPI("Used by Animation Events")]
         public void AnimEvent_Jump()
         {
-            if (TryPlayAnimEventFX(lastJumpTime, jumpIntervalSeconds, centerBottomTransform, AvatarAnimationEventType.Jump))
+            var movementState = GetMovementState();
+            if (movementState == MovementKind.None) return;
+
+            if (TryPlayAnimEventFX(lastJumpTime, jumpIntervalSeconds, centerBottomTransform, AvatarAnimationEventType.Jump,
+                    movementState switch
+                    {
+                        MovementKind.Jog => AvatarAudioClipType.JumpStartJog,
+                        MovementKind.Run => AvatarAudioClipType.JumpStartRun,
+                        _ => AvatarAudioClipType.JumpStartWalk
+                        }))
                 lastJumpTime = currentTime;
         }
 
         [PublicAPI("Used by Animation Events")]
         public void AnimEvent_Land()
         {
-            if (TryPlayAnimEventFX(lastLandTime, landIntervalSeconds, centerBottomTransform, AvatarAnimationEventType.Land))
-                lastLandTime = currentTime;
+            var movementState = GetMovementState();
+            if (movementState == MovementKind.None) return;
+
+            if (TryPlayAnimEventFX(lastLandTime, landIntervalSeconds, centerBottomTransform, AvatarAnimationEventType.Land,
+                    movementState switch
+                    {
+                        MovementKind.Jog => AvatarAudioClipType.JumpLandJog,
+                        MovementKind.Run => AvatarAudioClipType.JumpLandRun,
+                        _ => AvatarAudioClipType.JumpLandWalk
+                    })) { lastLandTime = currentTime; }
         }
 
         private void PlayStepSoundForFoot(Transform footTransform)
         {
             if (!AvatarAnimator.GetBool(AnimationHashes.GROUNDED)) return;
 
-            if (TryPlayAnimEventFX(lastFootstepTime, GetIntervalFor(GetMovementState()), footTransform, AvatarAnimationEventType.Step))
+            var movementState = GetMovementState();
+            if (movementState == MovementKind.None) return;
+
+            float interval = GetIntervalFor(movementState);
+
+            if (TryPlayAnimEventFX(lastFootstepTime, interval, footTransform, AvatarAnimationEventType.Step,
+                    movementState switch
+                    {
+                        MovementKind.Jog => AvatarAudioClipType.StepJog,
+                        MovementKind.Run => AvatarAudioClipType.StepRun,
+                        _ => AvatarAudioClipType.StepWalk,
+                    }))
```

Comment:
Looks like this logic block can be extracted, since its repeated three times. Its also kind of weird to sww a switch inside and if statemte

Reply:
Actually the returned AvatarAudioClipType is different for each kind of movementState (jump, step or land), but its true we can extract it to a more clean method that takes that info and produces the right AvatarAudioClipType. 
Nice :)